src/main/java/com/github/manolo8/darkbot/core/objects/facades/SlotBarsProxy.java

```diff
@@ -94,20 +91,38 @@ public boolean isProActionBarVisible() {
         return (!toggleProAction || settings.pressKeybind(SettingsProxy.KeyBind.TOGGLE_PRO_ACTION))
                 && settings.pressKeybind(SettingsProxy.KeyBind.of(slotBarType, slotNumber))
                 ? ItemUseResult.SUCCESS : ItemUseResult.FAILED;
+    }
 
+    //simply useItem method must check default flags.
+    private ItemUseResult checkItemFlags(boolean isUseItem, Item item, ItemFlag... flags) {
+        for (ItemFlag flag : flags) {
+            if (flag == ItemFlag.NONE) { // kinda may be not throw if none flag was passed last and test failed for another flag.
+                if (flags.length > 1)
+                    throw new IllegalArgumentException("You cannot pass NONE flag with any other!");
+                else if (!isUseItem) return null;
+            } else if (!flag.test(item))
+                return flag.getFailResult();
+        }
+
+        return !isUseItem ? null : Stream.of(DEFAULT_ITEM_FLAGS)
+                .filter(f -> f != ItemFlag.NONE)
+                .filter(f -> !f.test(item))
+                .map(ItemFlag::getFailResult)
+                .findFirst().orElse(null);
     }
```

Comment:
```suggestion
    private ItemUseResult checkItemFlags(Item item, ItemFlag... flags) {
        if (flags == null || flags.isEmpty()) flags = DEFAULT_ITEM_FLAGS;
        for (ItemFlag flag : flags) {
            if (flag == ItemFlag.NONE && flags.length > 1)
                    throw new IllegalArgumentException("ItemFlag.NONE can't be combined with other flag!");
            if (!flag.test(item)) return flag.getFailResult();
        }
        return null;
    }
```
I'm not exactly sure why the isUseItem parameter is even passed, if it's so that you can decide if you should add the default flags, just convert the first line in my suggestion in a return null, and always pass the default item flags from useItem method if the user doesn't set any.

Also, make the NONE flag simply a dummy test/result, so it can be treated just like any other without requiring a special case